### ISSUE TITLE
AddressTower

### ISSUE THREAD

**Issue by DaniellMesquita** [district0x/district-proposals#280]
## **Name**: 
AddressTower

## **Purpose**: 
A community-driven marketplace for registering and managing decentralized web domains.

## **Description**: 
District0x users will use their DNT tokens to propose decentralized domains with the extension .uuu (Universe Under Union, an evolution of the www. subdomain and meaning, also referring to IPFS - Inter Planetary File System), where the community will use their tokens to vote if this domain can be registered.

The proposed domain can also come with its DNS info, filled by the user who proposed it.

Main purpose of this community is to register humankind-valuable decentralized domains that can't be squatted for money; being different from Ethereum's ENS, Namecoin and Emercoin.

If the community approves that domain, it is registered and is property of AddressTower.
This brand new domain is locked to 1 year, so AddressTower can't expire it this mean time.

Users can register open-source or other humankind-valuable related domains, for free without fees, and that domain will be locked for 20 years. While locked, a domain can receive DNS update proposals from users, and receive voting. After the lock, the AddressTower community can vote to renew this domain. Personal domains are included on this category.

Domains related to other things will have a registry fee (and yearly fee), and will be locked for only 1 year.

Domains related to DAOs, that doesn't applies to the above, will have the lowest registry/yearly fees and locked for 5 years.

DNS entries will use JSON.
Unlike Namecoin/Emercoin, domains aren't case-sensitive.

## **Ethereum Address**:
0xDDfC2e10702d8A781727A34D83B3bb3CA94a3E91

**Comment by DaniellMesquita** [comment]
First domains:

district0x.uuu
aragon.uuu
pando.uuu
ipfs.uuu
ethereum.uuu
bitcoin.uuu
cjdns.uuu
althea.uuu
cryptocurrencies.uuu (informative site)
cryptocurrency.uuu (informative sites)
everipedia.uuu
dapp.uuu (informative sites)
dapps.uuu (informative sites)
stateofthedapps.uuu
linux.uuu
debian.uuu
beakerbrowser.uuu
dat.uuu
pfrazee.uuu
firefox.uuu
mozilla.uuu
colemak.uuu
qrl.uuu
tor.uuu

**Comment by Bradymck** [comment]
Hey, thanks for the proposal. So if this were to be voted into place as the next district, what technology would need to be written from the ground up, and what technology is already available for this? 

I'm having trouble seeing how to get this going as a marketplace without first creating an entirely new infrastructure to support this, similar to some of the other name infrastructure you mentioned.  Do you have some supplemental links or information we can read for this?

**Comment by DaniellMesquita** [comment]
@Bradymck
> Hey, thanks for the proposal. So if this were to be voted into place as the next district, what technology would need to be written from the ground up, and what technology is already available for this?

Both Aragon and Ethereum's smart contracts can be used for this.

> I'm having trouble seeing how to get this going as a marketplace without first creating an entirely new infrastructure to support this, similar to some of the other name infrastructure you mentioned. Do you have some supplemental links or information we can read for this?

This marketplace can use DNT tokens, or a brand new token (for AddressTower as a DAO/community) using Aragon's infrastructure.

About the name infrastructure, it can use Ethereum's smart contracts, whereas domains are property of the AddressTower DAO/marketplace, that are assets locked on the blockchain locked for some time according to the smart contract.

Sorry but I don't have exact links about using Aragon to create this DAO/tokens (but not a problem as District0x haves good communication with Aragon and already used their tech) and creating smart-contract(s) to comply with that DAO/marketplace for registering the domains.

Basically its just the use of Aragon and their SDKs, that interfaces with smart-contracts. Aragon already does haves all needed infrastructure to power AddressTower, which covers all the specs described on the issue's first comment

**Comment by Bradymck** [comment]
So to confirm, this would require development of a new name service in competition with Handshake, ENS, Name Coin, and Emercoin? 

If so, what inefficiency and problems do you see with these other systems that needs to be solved?

I will also add that Aragon and d0xINFRA aren't built to create new name services, it's primarily governance tools and standardized marketplace tools. 

So this new name service would need to be developed prior to this becoming a district because our framework would not be designed for developing this name service.

**Comment by DaniellMesquita** [comment]
@Bradymck

I'm pretty sure how this service is needed (beneficial for both people and District0x's image) and how d0x can help it, so I'll try my best to explain it and cover all your questions.

> So to confirm, this would require development of a new name service in competition with Handshake, ENS, Name Coin, and Emercoin?

That's a problem to have competition with existing products that haves their own philosophies that prevents them to upgrade how they works?
Because of Netscape (open competitor to IE) there came Firefox, and both Firefox and Netscape helped each other to build an better alternative browser; and now Firefox is very better than the latest IE and even better than Edge, which is a solution to an problem and because of that, I (and not only) consider Firefox an valuable project for the benefit of humankind.

All your referred domain registrars (Handshake/ICANN/ENS/Namecoin/Emercoin haves the same problem: money buys personal/open-source properties (as kaffiene.bit) and human-valuable brands (bitcoin.com) with domain squatting. As District0x haves focus on both money and open-source, AddressTower is where contributors receives tokens for their effort and users are secure on registering personal or humankind-valuable domains, while other kinds of domains haves a fee (not high as of ICANN) and valuable projects aren't registered for just 1 year for then be squatted.

There is no domain system like AddressTower (combining governing, security and decentralization), so this is very important for both users and also for District0x, which will be responsible for that.

> If so, what inefficiency and problems do you see with these other systems that needs to be solved?

Already explained above. But I know there are other problems that I didn't covered.

> I will also add that Aragon and d0xINFRA aren't built to create new name services, it's primarily governance tools and standardized marketplace tools.

Yes, AddressTower is both used for governance and a marketplace.

> So this new name service would need to be developed prior to this becoming a district because our framework would not be designed for developing this name service.

The only new thing to be done is using an already existing technology: as there are tokens that are owned by an DAO (and not by an specific ETH address - already covered by Aragon), domains will also be tokens owned by an DAO, but non-fungible tokens (already existing ERCs for that).

**Comment by Bradymck** [comment]
Thanks for the quick response! I like where your head is at on this. I can absolutely see room for improvement in ENS and other similar systems. Possibly even another alternative, hard to say this early without seeing a better alternative market emerge first though. 

A lot of people are disappointed with the direction that ENS has been going too. I do however, think it's too early to say that Aragon couldn't be implemented to solve those community governance issues you raise. 

I may also add that I don't see it out of the realm of possibility that a popular ENS marketplace gains influence over ENS development in the future. I hope it's Name Bazaar, but I may be biased 😉

It's a bit early to say but definitely possible if the Name Bazaar (or another) community put their money where their mouth is and starting self funding bounties, forking code, and contributing from a successful/popular decentralized governance entity. 

> **There is no domain/TLD/registrar/DNS system like AddressTower** 

My disconnect on how this could be deployed as a marketplace is the lack of an ERC token standard for this use case. Before we could feasibly deploy this as a district, there would have to already be an established market built around that standard.

**Examples:**
People already use ENS, so the Name Bazaar marketplace was a clear option we could iterate upon.  

Another example would be how Ethlance iterates upon an already established freelancing market. 

While Meme Factory iterates upon an already established market for ERC721 tokens and digital collectibles as a whole. 
 
On a larger scale, Uber took an already established taxi industry and iterated upon it, to create a more open and inclusive marketplace. 

So using our tools and Aragon makes sense to make and Uber alternative better but they don't establish that market framework so it can be improved upon.  

> **So this is very important for both users and also for District0x, which will be responsible for that.**

Considering the fact that an entirely new token standard first needs to be developed, it's almost akin to building an entire taxi industry so you can iterate upon that marketso you can then build a ride sharing marketplace. 

I see it as putting to "cart before the horse" so to speak. Don't get me wrong, if you wanted to go build this ERC standard, establish the market and use our framework to power it, it could save you a lot of time. All you have to do is find a way to fund and develop that new token standard. 

Me personally, I would like to see the ENS community start using Name Bazaar more and start donating their proceeds to an aragon entity that they all fund ENS development out of. 

This could go a long way to showing how Aragon can be used as a way to influence open source development.  

On a total side note, I'm reading 'Zero to One' right now and want to say that your line of thinking is exactly the type of thinking that Peter Thiel praises and encourages in the book. 

"Doing something no one else is doing, rather than iterating upon already established ideas"

If You are already working on a new ERC token standard for this use case, please let me know. 

Though I do need to close this proposal out unless you, or someone else is already developing the market and standards for this idea. 

If this won a vote, the reality is that we would have to divert all our development resources to an entirely new initiative outside the scope of district0x just to kick start this new standard. 

Leaving most of what we are developing behind. Because this is quite an initiative. Which we just don't have the ability to do so at this time. 

That being said, that may change as we start to onboard more third party development teams and launch the district creation interface, but that is quite a ways away. Let me know if you are building a team to do this, I would be happy to schedule a call to see how we can integrate your new initiative with our tools and where I see room for collaboration. If so I could also reopen this proposal if you are taking that kind of initiative. 

Also, you will still get the proposal reward for this. Just an FYI. I like where your head is at for sure. 😃

**Comment by DaniellMesquita** [comment]
@Bradymck
> Thanks for the quick response! I like where your head is at on this.

> Also, you will still get the proposal reward for this. Just an FYI. I like where your head is at for sure. :smiley: 

From the first to the last paragraph, "where your head is" remembers what a user said on two days before: "Take your head out of your ass". What did you mean? Coincidence? (ref. https://github.com/aragon/nest/issues/125#issuecomment-451688185)

> I can absolutely see room for improvement in ENS and other similar systems.

How can ENS being improve without totally restructuring it?

> Possibly even another alternative,

That's why I've suggested AddressTower.

> hard to say this early without seeing a better alternative market emerge first though.

Bad for district0x if it just waits for other's innovation instead of being innovative; the idea of AddressTower is already here.

> A lot of people are disappointed with the direction that ENS has been going too. I do however, think it's too early to say that Aragon couldn't be implemented to solve those community governance issues you raise.

.eth domains are relative to Ethereum network, while .uuu domains represents universal domains for a common-good; and I doubt ENS will accept a new TLD. Also, when being governed by DAO, ENS would need a total restructure.

> I may also add that I don't see it out of the realm of possibility that a popular ENS marketplace gains influence over ENS development in the future. I hope it's Name Bazaar, but I may be biased :wink: 

"I don't see possibility that a popular ENS marketplace gains influence over ENS development"

Like Ethereum didn't influenced on Bitcoin, as they haves their own direction; that's the purpose of AddressTower.

> It's a bit early to say but definitely possible if the Name Bazaar (or another) community put their money where their mouth is and starting self funding bounties, forking code, and contributing from a successful/popular decentralized governance entity.

That's just a frontend of ENS, right? So I don't know how they're related to AddressTower.

> "There is no domain/TLD/registrar/DNS system like AddressTower"
> 
> My disconnect on how this could be deployed as a marketplace is the lack of an ERC token standard for this use case.

You're making me repeat myself again:
> The only new thing to be done is using an already existing technology: as there are tokens that are owned by an DAO (and not by an specific ETH address - already covered by Aragon), domains will also be tokens owned by an DAO, but non-fungible tokens (already existing ERCs for that).

ERC721 can power decentralized domains plus the governance model of Aragon.

> Before we could feasibly deploy this as a district, there would have to already be an established market built around that standard.

So district0x isn't innovative, which is bad for its advertisement/marketing.
There is already a established market for decentralized DNS, and these consumers are looking for more alternatives. AddressTower is currently the best workaround for governed yet decentralized, human-valuable "ICANN".



> Examples:
> People already use ENS, so the Name Bazaar marketplace was a clear option we could iterate upon.

Already explained the limitations of ENS, and how district0x could innovate on that.

> Another example would be how Ethlance iterates upon an already established freelancing market.

Yes, as AddressTower should iterate on the already established market of decentralized domains; but innovation is the key of good projects and that doesn't looks for established market (as Bitcoin and Ethereum didn't looked for that, but innovated without the mass believing on they).

> While Meme Factory iterates upon an already established market for ERC721 tokens and digital collectibles as a whole.

AddressTower does the same when using domains as ERC721 tokens, as explained above. 

> On a larger scale, Uber took an already established taxi industry and iterated upon it, to create a more open and inclusive marketplace.

Please stop the rhetorical.
Already explained above, and on my previous comments, but you kept not understanding the purpose and the benefit for District0x.

> So using our tools and Aragon makes sense to make and Uber alternative better but they don't establish that market framework so it can be improved upon.

Same for decentralized DNS.

> So this is very important for both users and also for District0x, which will be responsible for that.

Because of that, AddressTower is beneficial for District0x.

> Considering the fact that an entirely new token standard first needs to be developed,

I didn't said that.

> it's almost akin to building an entire taxi industry so you can iterate upon that marketso you can then build a ride sharing marketplace.

AddressTower doesn't builds a entire decentralized domain industry, but uses the existing. Same for the ERC standard that AddressTower doesn't needs to create but iterate upon it (ERC721).

> I see it as putting to "cart before the horse" so to speak.

Who told you to say that?
No need to tell.

> Don't get me wrong, if you wanted to go build this ERC standard, establish the market and use our framework to power it, it could save you a lot of time.

As said, there is already an ERC standard for that, and the decentralized TLD/DNS market is already established.

> All you have to do is find a way to fund and develop that new token standard.
> 
> Me personally, I would like to see the ENS community start using Name Bazaar more and start donating their proceeds to an aragon entity that they all fund ENS development out of.
> 
> This could go a long way to showing how Aragon can be used as a way to influence open source development.

If ENS began to be governed by community, accept human-valuable domains that can't be squatted and accepts new domain extensions. If AddressTower does that, it will be an amazing advancement and calling traction to District0x.

> On a total side note, I'm reading 'Zero to One' right now and want to say that your line of thinking is exactly the type of thinking that Peter Thiel praises and encourages in the book.
> 
> "Doing something no one else is doing, rather than iterating upon already established ideas"

That is what District0x needs.

> If You are already working on a new ERC token standard for this use case, please let me know.
> 
> Though I do need to close this proposal out unless you, or someone else is already developing the market and standards for this idea.

So you can re-open, as the ERC is already here and is a established market, like decentralized domains is also.

> If this won a vote, the reality is that we would have to divert all our development resources to an entirely new initiative outside the scope of district0x just to kick start this new standard.

The standard is already here. If AddressTower were not related to District0x, I shouldn't propose it here. But I'm sure AddressTower is an innovative market and beneficial for District0x. No need to divert the development resources, as AddressTower needs the same kind of work as the other markets.

> Leaving most of what we are developing behind.

You could postpone it or balance with other existing projects, as you can see how AddressTower is needed for both users and your ecosystem.

> Because this is quite an initiative. Which we just don't have the ability to do so at this time.

That is just like the other marketplaces that haves their own difference/characteristics.

> That being said, that may change as we start to onboard more third party development teams and launch the district creation interface, but that is quite a ways away. Let me know if you are building a team to do this, I would be happy to schedule a call to see how we can integrate your new initiative with our tools and where I see room for collaboration. If so I could also reopen this proposal if you are taking that kind of initiative.

Above is already explained about that.

**Comment by Bradymck** [comment]
> The idea of AddressTower is already here. (And in regards to your feedback and other comments.) 

I'm not saying this couldn't or shouldn't be done. On the contrary, it is already being done with ENS. People are converting ENS names into ERC721 tokens as we speak.

Though this is not so much a new marketplace as it is an extension of ENS in my view so it could use some research on how other name services could use ERC721 as a universal standard for many name types. New ones like your idea included.

---

However to clarify what we're building, district0x is a framework for community marketplaces to operate as a connected network of user governed marketplaces, along with all the features in place to allow third party expansion of the network. It's not meant as an incubator fornew projects like this. An endeavor like this would require dedicated funding and a development team among other things. 

So building an entirely new name service is outside our roadmap and outside the scope of what were building in the near term. After the network is fully deployed and more third party dev teams are on board using our SDK I could see that changing but it's a bit early for that. 

That being said, the tools we're deploying that will allow third party expansion of the network are already seeing many different projects come on board. 

To name a few things the teams are building 👉🏽 identity systems, media services, marketing and design, decentralized oracle networks, game developers, and many others. 

So as these third party development studios begin integrating with our SDK and building their marketplaces with our tools, I can imagine a team focused on building alternatives to ENS could do well too. 

I will reopen but if you could do me a favor and get a roadmap, to do list, and issues section for research. You may also want to start rounding up people interested in supporting development. 

In current form, this idea needs a lot of research because implementing it is not straightforward until some key developments are made. 

Also, this will be something you need to build an MVP for to get the community excited and vote for it. 

If you did get an MVP made I could see it gaining traction within the community quite quickly. Please do drop by Discord some time to chat in real time about this. I'm sure others would love to give feedback and ideas.  https://discord.gg/P9RQejv

> From the first to the last paragraph, "where your head is" remembers what a user said on two days before: "Take your head out of your ass". What did you mean? Coincidence? (ref. aragon/nest#125 (comment))

I had not seen these comments. This person is terribly negative ☹. I meant that I genuinely like your line of thinking. It's a good idea but needs a lot of work to make it happen. I was not referencing this conversation with that person. That person should be removed after speaking to you that way. Ideas should not be met with ridicule. They should be challenged so they can be fine tuned. Not shot down with negativity.

**Comment by Bradymck** [comment]
Reopened

**Comment by Bradymck** [comment]
Please do get a roadmap along with all resources that could help build this.

**Comment by Bradymck** [comment]
FYI, I will be digging into this proposal on stream today. Please do drop by if you want to answer any questions people may have about it.

**Comment by DaniellMesquita** [comment]
Peep about this: https://peepeth.com/daniellmesquita/peeps/QmTUH32fVEc5awr86CB7jpn6Fc2S4W43XXWZBXVMNw4AtQ

**Comment by DaniellMesquita** [comment]
> Please do get a roadmap along with all resources that could help build this.

I'm working around it by myself using IPNS and DAO to a json registry till it begins using ERC models.
